substrate/client/cli/src/params/network_params.rs

```diff
@@ -246,6 +245,18 @@ impl NetworkParams {
 				is_dev || matches!(chain_type, ChainType::Local | ChainType::Development),
 		};
 
+		// `IdentifyVariant` trait is defined in polkadot, however we need this bit of information
+		// in the network configuration.
+		let is_kusama = chain_spec.id().starts_with("kusama") || chain_spec.id().starts_with("ksm");
+
+		// If the network backend was specified, nothing is changed.
+		// Kusama defaults to litep2p, while other chains to libp2p.
+		let network_backend = if is_kusama {
+			self.network_backend.unwrap_or(NetworkBackendType::Litep2p)
+		} else {
+			self.network_backend.unwrap_or(NetworkBackendType::Libp2p)
+		};
+
```

Comment:
This needs to be moved to the polkadot service.

Reply:
dq/nit: would it better to check genesis hash?

Reply:
Depends as you probably also want this for your local kusama network, using the id is fine.